gateway-ha/src/test/java/io/trino/gateway/ha/handler/TestQueryIdCachingProxyHandler.java

```diff
@@ -33,20 +34,20 @@ public class TestQueryIdCachingProxyHandler
     public void testExtractQueryIdFromUrl()
             throws IOException
     {
-        String[] paths = {
-                "/ui/api/query/20200416_160256_03078_6b4yt",
-                "/ui/api/query/20200416_160256_03078_6b4yt?bogus_fictional_param",
-                "/ui/api/query?query_id=20200416_160256_03078_6b4yt",
-                "/ui/api/query.html?20200416_160256_03078_6b4yt"};
-        for (String path : paths) {
-            String queryId = QueryIdCachingProxyHandler.extractQueryIdIfPresent(path, null);
+        URI[] uris = {
+                URI.create("http://example.com/ui/api/query/20200416_160256_03078_6b4yt"),
+                URI.create("http://example.com/ui/api/query/20200416_160256_03078_6b4yt?bogus_fictional_param"),
+                URI.create("http://example.com/ui/api/query?query_id=20200416_160256_03078_6b4yt"),
+                URI.create("http://example.com/ui/api/query.html?20200416_160256_03078_6b4yt")};
```

Comment:
also .. will the path really always be ui/api/query .. and we depend on that.. if not .. should we add some other valid ones?

Reply:
a few other cases with `https` and `www.example.com` would be good.

Reply:
How about adding other tests for these apis without queryid?
- ui/api/cluster
- ui/api/query
- ui/api/stats

Reply:
@Chaho12 are these negative tests to add under `nonMatchingUris`?

Reply:
I was thinking more about how it will route in such cases, then i just realized that test only checks if it extracts queryid or not.
so sure, adding it at nonMatchingUris is good (but now i don't think it's much meaningful test case though)